src/dfm/setup.rs

```diff
@@ -0,0 +1,25 @@
+use super::model::Config;
+use anyhow::{Result, anyhow};
+use std::fs;
+use std::path::PathBuf;
+
+pub fn setup(path: PathBuf) -> Result<()> {
+    let config = generate_config(path)
+
+    println!("{:?}", config?);
+
+    Ok(())
+}
+
+fn generate_config(path: PathBuf) -> Result<Config> {
+    let config_content: String = fs::read_to_string(&path).expect("");
```

Comment:
```suggestion
    let config_content: String = fs::read_to_string(&path).with_context(|| format!("Cannot read file {path}"))?;
```
Expect will panic. Since you're already in a function that returns `Result` my advice is to add context to the result and early return. I think this is how you do it but also I didn't try to compile this suggestion so YMMV. Check out the [anyhow docs](https://docs.rs/anyhow/latest/anyhow/trait.Context.html) for more info. Mainly I think using the `?` at the end of a result type is what you want.